taichi/ui/backends/vulkan/renderable.cpp

```diff
@@ -49,22 +49,45 @@ void Renderable::update_data(const RenderableInfo &info) {
   }
 
   int num_vertices = info.vbo.shape[0];
+  int draw_num_vertices = info.draw_vertex_count;
+  int draw_first_vertices = info.draw_first_vertex % num_vertices;
+
   int num_indices;
+  int draw_num_indices;
+  int draw_first_indices;
+
   if (info.indices.valid) {
     TI_ERROR_IF(info.indices.matrix_cols != 1,
                 "indices must either be a ti.field or a 2D/3D ti.Vector.field");
     num_indices = info.indices.shape[0] * info.indices.matrix_rows;
+    draw_num_indices = info.draw_index_count * info.indices.matrix_rows;
+    draw_first_indices =
+        (info.draw_first_index * info.indices.matrix_rows) % num_indices;
     if (info.indices.dtype != PrimitiveType::i32 &&
         info.indices.dtype != PrimitiveType::u32) {
       throw std::runtime_error("dtype needs to be 32-bit ints for indices");
     }
   } else {
     num_indices = 1;
+    draw_num_indices = 1;
+    draw_first_indices = 0;
   }
 
   config_.vertices_count = num_vertices;
   config_.indices_count = num_indices;
 
+  if (info.has_user_customized_draw) {
```

Comment:
Can we remove this flag and assign to `config_.draw_xxx` directly?

Reply:
No, I didn't use this at the first time, but I found that other part such as line, triangle, circle and so on were not need for this function at present. So I must use this bool value to close this control for they all were draw as renderable. (May lead to bugs when draw lines, triangles and circles)